### DISCUSSION TITLE
Shared tests for multiple websites/repos

### DISCUSSION THREAD

**Discussion by idevelopbugs** [cypress-io/cypress#16225] [Component Testing]
Hi!

I'm trying to figure out the best way to reuse tests across multiple websites. We have a big platform with react components and import these into the sites. But each web has it's own cypress setup with testscripts, and they pretty much look the same.

The ideal scenario would be to have 1 repo with tests which could easily be imported to each web. Im thinking about maybe having all tests in a repo, then importing them as "commands", and then  just executing them in each web by `cy.platformTest('someSelector')`. But it still feels like there should be a better way. Any ideas?

**Comment by lmiller1990** [comment]
Do you want to reuse actual *tests* (like `.spec.js`) or just some shared code?

The way I've handled this is custom commands (or custom functions imported to `support/index.js`. We had something like `@my-org/cypress-shared` might contain a `commands.js`:

```js
Cypress.command.add('clickOnButton', text => {
  cy.get('button').contains(text).click()
})
```

Then just imported this to our `support/index.js`:

```js
// shared commands
import '@my-org/cypress-shared/commands.js'
```

Is this what you mean? I'm not sure there is anything Cypress specific about sharing code.

**Reply by idevelopbugs** [reply]
Thanks for the reply! I was thinking about reusing actual tests and importing them from an outside repo and not just "helper functions" as the ones typically used in commands.

**Reply by bahmutov** [reply]
Not sure what you mean by "reusing actual tests", something like https://www.cypress.io/blog/2019/12/06/use-meaningful-smoke-tests/ ?
